src/Microsoft.VisualStudio.Editors/PropPages/ReferencePropPage.vb

```diff
@@ -788,31 +788,11 @@ Namespace Microsoft.VisualStudio.Editors.PropertyPages
                 pCallback:=cancellationCallback)
 
             Try
-                Dim componentModel = DirectCast(ServiceProvider.GetService(GetType(SComponentModel)), IComponentModel)
-                Dim visualStudioWorkspace = componentModel.GetService(Of VisualStudioWorkspace)
-                Dim solution = visualStudioWorkspace.CurrentSolution
-                Dim CurrentImportNames As New List(Of String)
-
-                For Each project In solution.Projects
-
-                    ' We need to find the project that matches by project file path
-                    If project.FilePath IsNot Nothing AndAlso String.Compare(project.FilePath, DTEProject.FullName, ignoreCase:=True) = 0 Then
-                        Dim compilationTask = project.GetCompilationAsync(cancellationTokenSource.Token)
-                        compilationTask.Wait(cancellationTokenSource.Token)
-                        Dim compilation = compilationTask.Result
-                        Dim compilationOptions = CType(compilation.Options, VisualBasicCompilationOptions)
-                        If (compilationOptions IsNot Nothing) Then
-                            For Each globalImport In compilationOptions.GlobalImports
-                                CurrentImportNames.Add(globalImport.Name)
-                            Next
-                            CurrentImportNames.Sort(CaseInsensitiveComparison.Comparer)
-                        End If
-                        Return CurrentImportNames.ToArray()
-                    End If
-                Next
-
-                ' Return empty list if an error occurred
-                Return New String() {}
+                Dim vsImports As VSLangProj.Imports = CType(DTEProject.Object, VSLangProj.VSProject).Imports
+                Dim result As New List(Of String)(vsImports.Count)
```

Comment:
Nit, can't you just pass the imports direct to the constructor?

Reply:
Or just use Linq completely to sort/ToArray.

Reply:
Constructor takes either capacity or IEnumerable, and I figured it wouldn't presize with the IEnumerable approach. (Not sure if the type tests for count see through COM enumerables).  Could definitely just switch to an OrderBy though if you want.